JAICore/jaicore-ml/test/jaicore/ml/dyadranking/general/AdvancedDyadDatasetDyadRankerTester.java

```diff
@@ -41,30 +35,66 @@ public AdvancedDyadDatasetDyadRankerTester(ADyadRanker ranker) {
 
 	@Before
 	public void trainRanker() throws TrainingException {
-		ranker.train(DyadRankingInstanceSupplier.getDyadRankingDataset(100, 100));
+		ranker.train(DyadRankingInstanceSupplier.getDyadRankingDataset(10, 1000));
 	}
 
 	@Test
 	public void testSwapOrdering1() throws PredictionException {
 		System.out.println("Now testing ordering");
-
-		IDyadRankingInstance test = DyadRankingInstanceSupplier.getDyadRankingInstance(100);
-		IDyadRankingInstance predict = ranker.predict(test);
-
-		Dyad currentMin = predict.getDyadAtPosition(0);
-		int failures = 0;
-		for (Dyad dyad : predict) {
-			if (!(DyadRankingInstanceSupplier.complexDyadRanker().compare(currentMin, dyad) <= 0)) {		
-				failures++;
-			} 
-			currentMin = dyad;
+		
+		int maxDyadRankingLength = 10;
+		int nTestInstances = 100;
+		double avgKendallTau = 0;
+		double avgFailures = 0;
+		
+		for (int testInst = 0; testInst < nTestInstances; testInst++) {
+			IDyadRankingInstance test = DyadRankingInstanceSupplier.getDyadRankingInstance(maxDyadRankingLength);
+			IDyadRankingInstance predict = ranker.predict(test);
+			
+			int dyadRankingLength = test.length();
+			int nConc = 0;
+			int nDisc = 0;
+			for (int i = 1; i < dyadRankingLength; i++) {
+				for (int j = 0; j < i; j++) {
+					if (DyadRankingInstanceSupplier.complexDyadRanker().compare(
+							predict.getDyadAtPosition(j), predict.getDyadAtPosition(i)) < 0) {
+						nConc++;
+					} else {
+						nDisc++;
+					}
+				}
+			}
+			double kendallTau = 2.0 * (nConc - nDisc) / (dyadRankingLength * (dyadRankingLength - 1) );
+			avgKendallTau += kendallTau;
+			
+			Dyad currentMin = predict.getDyadAtPosition(0);
+			int failures = 0;
+			for (Dyad dyad : predict) {
+				if (!(DyadRankingInstanceSupplier.complexDyadRanker().compare(currentMin, dyad) <= 0)) {		
+					failures++;
+				} 
+				currentMin = dyad;
+			}
+			avgFailures += failures;
 		}
-		System.out.println("Found failures: "+ failures);
+		avgKendallTau /= nTestInstances;
+		avgFailures /= nTestInstances;
+		
+		System.out.println("Kendall's tau: " + avgKendallTau); 
```

Comment:
We have a class that is made to use multilabel measures that are already implemented for the context of dyad ranking: 
DyadRankingMLLossFunctionWrapper in jaicore.ml.dyadranking.loss can use all the ADecomposableDoubleMeasures from jaicore.ml.core.evaluation.measure.multilabel so if we add a measure here maybe it should be added as a multilabel measure? + the already implemented measures could possibly be used here for additional output.

Reply:
Should definitely be done in the future. However, the class does not exist on this branch yet.